include/event_manager/event_manager.h

```diff
@@ -0,0 +1,317 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+
+/** @file
+ * @brief Event manager header.
+ */
+
+#ifndef ZEPHYR_INCLUDE_EVENT_MANAGER_EVENT_MANAGER_H_
+#define ZEPHYR_INCLUDE_EVENT_MANAGER_EVENT_MANAGER_H_
+
+
+/**
+ * @defgroup event_manager Event Manager
+ * @brief Event Manager
+ *
+ * @{
+ */
+
+#include <zephyr.h>
+#include <zephyr/types.h>
+#include <sys/reboot.h>
+#include <sys/__assert.h>
+#include <logging/log_ctrl.h>
+
+#include <event_manager/event_manager_priv.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+
+/** @def SUBS_PRIO_MIN
+ *
+ * @brief Index of the highest subscriber priority level.
+ */
+#define SUBS_PRIO_MIN    _SUBS_PRIO_FIRST
+
+
+/** @def SUBS_PRIO_MAX
+ *
+ * @brief Index of the lowest subscriber priority level.
+ */
+#define SUBS_PRIO_MAX    _SUBS_PRIO_FINAL
+
+
+/** @def SUBS_PRIO_COUNT
+ *
+ * @brief Number of subscriber priority levels.
+ */
+#define SUBS_PRIO_COUNT (SUBS_PRIO_MAX - SUBS_PRIO_MIN + 1)
+
+
+/** @brief Event header.
+ *
+ * When defining an event structure, the event header
+ * must be placed as the first field.
+ */
+struct event_header {
+	/** Linked list node used to chain events. */
+	sys_snode_t node;
+
+	/** Pointer to the event type object. */
+	const struct event_type *type_id;
+};
+
+
+/** @brief Dynamic event data.
+ *
+ * When defining an event structure, the dynamic event data
+ * must be placed as the last field.
+ */
+struct event_dyndata {
+	/** Size of the dynamic data. */
+	size_t size;
+
+	/** Dynamic data. */
+	uint8_t data[0];
+};
+
+
+/** @brief Event listener.
```

Comment:
do we need `event_listener` `event_subscriber` and `event_type` to be in public header? Maybe they could be moved to priv header?